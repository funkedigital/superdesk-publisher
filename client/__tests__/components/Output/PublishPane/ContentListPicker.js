import React from "react";
import ContentListPicker from "../../../../components/Output/PublishPane/ContentListPicker";
import { render } from "@testing-library/react";

const destination = {
  tenant: {
    code: 'tenant1',
    content_lists: [
      { type: "manual", id: 1, name: 'list1', content_list_items_count: 10 },
      { type: "automatic", id: 2, name: 'listAutomatic' },
      { type: "manual", id: 3, name: 'list3', content_list_items_count: 2 },
      { type: "manual", id: 4, name: 'list4', content_list_items_count: 0 }
    ]
  },
  content_lists: [
    { id: 1, position: 0 },
    { id: 3, position: 0 }
  ]
};

describe("Output/PublishPane/ContentListPicker", () => {
  it("renders properly", async () => {
    const { container } = render(
      <ContentListPicker
        destination={destination}
        update={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("remainingLists are filtered correctly", async () => {
    const { getByText } = render(
      <ContentListPicker
        destination={destination}
        update={jest.fn()}
      />
    );

    expect(getByText('Add to content list')).toBeInTheDocument();
  });
});
