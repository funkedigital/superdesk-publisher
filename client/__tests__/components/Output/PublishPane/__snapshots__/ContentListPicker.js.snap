// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/PublishPane/ContentListPicker renders properly 1`] = `
<div
  class="sd-list-item sd-list-item--no-hover sd-margin-b--1"
>
  <div
    class="sd-list-item__column sd-list-item__column--grow"
  >
    <div
      class="sd-list-item__row sd-list-item__row--only-child"
    >
      <span
        class="sd-margin-r--1"
      >
        Choose list
      </span>
      <div
        class="sd-line-input sd-line-input--is-select sd-list-item--element-grow sd-line-input--no-margin"
      >
        <select
          class="sd-line-input__select"
        >
          <option
            value="1"
          >
            list1
          </option>
          <option
            disabled=""
            value="3"
          >
            list3
          </option>
          <option
            value="4"
          >
            list4
          </option>
        </select>
      </div>
      <a
        class="icn-btn disabled"
        flow="left"
        sd-tooltip="Remove list"
      >
        <i
          class="icon-trash"
        />
      </a>
    </div>
    <div
      class="sd-list-item__row sd-margin-b--1"
    >
      <span
        class="sd-margin-r--1"
      >
        Set order
      </span>
      <div
        class="sd-line-input sd-line-input--is-select sd-list-item--element-grow sd-line-input--no-margin sd-line-input--no-label sd-margin-r--2"
      >
        <select
          class="sd-line-input__select"
        >
          <option
            value="0"
          >
            1
          </option>
          <option
            value="1"
          >
            2
          </option>
          <option
            value="2"
          >
            3
          </option>
          <option
            value="3"
          >
            4
          </option>
          <option
            value="4"
          >
            5
          </option>
          <option
            value="5"
          >
            6
          </option>
          <option
            value="6"
          >
            7
          </option>
          <option
            value="7"
          >
            8
          </option>
          <option
            value="8"
          >
            9
          </option>
          <option
            value="9"
          >
            10
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
